kubernetes-ingress-defaultbackend.yaml

```diff
@@ -38,7 +38,18 @@ update:
     strip-prefix: v
 
 test:
+  environment:
+    contents:
+      packages:
+        - wait-for-it
+        - curl
   pipeline:
     - name: help message
       runs: |
         defaultbackend --help
+    - name: Basic functional test for endpoints
+      runs: |
+        defaultbackend &
+        wait-for-it localhost:8080 -t 10
+        curl -sI --retry 5 http://localhost:8080/ | grep "404 Not Found"
```

Comment:
We don't need to validate the response was what we wanted with grep. Do `curl -f -LI...`

Reply:
@EyeCantCU if we use `curl -f -LI` the -f option causes curl to fail on HTTP errors. However, in our case, the default backend is designed to return a 404 Not Found for the root endpoint (/), which is the expected behavior using `curl -f` here cause the test to fail upon receiving the expected 404 response. Therefore, I've opted to use `curl -LI` without the -f option for the root endpoint, allowing the test to pass when a 404 is correctly returned.